llvm/lib/SYCLLowerIR/CompileTimePropertiesPass.cpp

```diff
@@ -97,6 +100,63 @@ MDNode *buildSpirvDecorMetadata(LLVMContext &Ctx, uint32_t OpCode,
   return MDNode::get(Ctx, MD);
 }
 
+/// Builds a metadata node for a SPIR-V decoration for cache controls
+/// where decoration code and value are both uint32_t integers.
+/// The value encodes a cache level and a cache control type.
```

Comment:
Is there a design document or specifications for LLVM/SPIR-V levels?
Cache controls are using annotated pointer design. Right?

Reply:
The SPIR-V extension is here: https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/INTEL/SPV_INTEL_cache_controls.asciidoc

This implementation builds upon annotated_ptr by adding new properties.

Reply:
Thanks. I also find https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/2140/, which documents the LLVM IR representation of cache controls.